R/zchunk_L222.en_transformation_USA.R

```diff
@@ -67,210 +75,638 @@ module_gcam.usa_L222.en_transformation_USA_DISABLED <- function(command, ...) {
+      repeat_add_columns(tibble(state = gcamusa.STATES)) %>%
+      filter((subsector == "oil refining" & state %in% oil_refining_states) |
+               subsector != "oil refining") %>%
+      mutate(technology = paste(state, subsector, sep = gcamusa.STATE_SUBSECTOR_DELIMITER)) ->
+      L222.Tech_USAen
+
+
+    # save some of this information for the PassThroughSector information
+    # L222.PassThroughSector_USAen: PassThroughSector information to send vintaging info from states to USA.
+    L222.Tech_USAen %>%
+      select(state, subsector, supplysector, region) %>%
+      rename(marginal.revenue.market = region,
+             region = state,
+             pass.through.sector = subsector,
+             marginal.revenue.sector = supplysector) ->
+      L222.PassThroughSector_USAen
+
+
+    # select only relevant columns for L222.Tech_USAen, particularly dropping state
+    L222.Tech_USAen %>%
+      select(one_of(LEVEL2_DATA_NAMES[["Tech"]])) ->
+      L222.Tech_USAen
+
+
+    # L222.TechInterp_USAen: technology shareweights, USA region
+    # Technology interpolation only applies to calibrated technologies.
+    # For biomass liquids, allow state shares to shift over time
+    # (future techs are different than present techs).
+    L222.Tech_USAen %>%
+      filter(subsector %in% c("oil refining", "biomass liquids")) %>%
+      mutate(apply.to = "share-weight",
+             from.year = max(BASE_YEARS),
+             to.year = max(MODEL_YEARS),
+             interpolation.function = if_else(subsector == "biomass liquids", "s-curve","fixed")) ->
+      L222.TechInterp_USAen
+
+
+    # L222.TechShrwt_USAen: technology shareweights in each year, USA region
+    # Default the base year shareweights to 0. This will be over-ridden in calibration
+    # Default the future year shareweights to 1.
+    L222.Tech_USAen %>%
+      repeat_add_columns(tibble(year = BASE_YEARS)) %>%
+      mutate(share.weight = 0) ->
+      tmp
+
+    L222.Tech_USAen %>%
+      repeat_add_columns(tibble(year = FUTURE_YEARS)) %>%
+      mutate(share.weight = 1) %>%
+      bind_rows(tmp) ->
+      L222.TechShrwt_USAen
+
+
+    # L222.TechCoef_USAen: technology coefficients and market names, USA region
+    L222.TechShrwt_USAen %>%
+      select(one_of(LEVEL2_DATA_NAMES[["TechYr"]])) %>%
+      mutate(minicam.energy.input = subsector,
+             coefficient = 1,
+             market.name = technology) %>%
+      separate(market.name, c("market.name", "trash"), extra = "merge", sep = gcamusa.STATE_SUBSECTOR_DELIMITER) %>%
+      select(-trash) ->
+      L222.TechCoef_USAen
+
+
+    # L222.Production_USArefining: calibrated refinery production in USA (consuming output of states)
+    # Aggregated to the supplysector/subsector/technology level
+    ### I'm pretty sure all this aggregation does is remove the state column, since the state information is
+    ### implicit in the technology column. That is, the aggregation doesn't actually change any values - is
```

Comment:
If that step doesn't do anything, let's remove it.

Reply:
I'm tempted to leave it as is in case we do something more dynamic in the future?

Reply:
it's also possible that that is a mistake and not the intended behavior